pkg/api/customization/project/project_actions.go

```diff
@@ -0,0 +1,147 @@
+	switch actionName {
+	case "setpodsecuritypolicytemplate":
+		return h.setPodSecurityPolicyTemplate(actionName, action, apiContext)
+	}
+
+	return errors.Errorf("bad action %v", actionName)
+}
+
+func (h *Handler) setPodSecurityPolicyTemplate(actionName string, action *types.Action, request *types.APIContext) error {
+	actionInput, err := parse.ReadBody(request.Request)
+	if err != nil {
+		return err
+	}
+
+	schema := request.Schemas.Schema(&managementschema.Version, client.PodSecurityPolicyTemplateProjectBindingType)
+	if schema == nil {
+		return fmt.Errorf("no %v store available", client.PodSecurityPolicyTemplateProjectBindingType)
+	}
+
+	binding, err := h.createOrUpdateBinding(request, schema, actionInput)
+	if err != nil {
+		return err
+	}
+
+	err = h.updateProjectPSPTID(request, actionInput)
+	if err != nil {
+		return err
+	}
+
+	request.WriteResponse(http.StatusOK, binding)
+
+	return nil
+}
+
+func (h *Handler) createOrUpdateBinding(request *types.APIContext, schema *types.Schema,
+	actionInput map[string]interface{}) (map[string]interface{}, error) {
+	bindings, err := schema.Store.List(request, request.Schema, &types.QueryOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var binding map[string]interface{}
+	for _, candidate := range bindings {
+		if candidate["projectId"] == request.ID {
+			binding = candidate
+			break
+		}
+	}
+
+	if binding == nil {
+		binding, err = h.createNewBinding(request, schema, actionInput)
+		if err != nil {
+			return nil, fmt.Errorf("error creating binding: %v", err)
+		}
+	} else {
+		binding, err = h.updateBinding(binding, request, schema, actionInput)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return binding, nil
+}
+
+func (h *Handler) updateProjectPSPTID(request *types.APIContext, actionInput map[string]interface{}) error {
+	project, err := request.Schema.Store.ByID(request, request.Schema, request.ID)
+	if err != nil {
+		return fmt.Errorf("error getting current project: %v", err)
+	}
+
+	status, ok := project["status"]
+	if !ok {
+		project["status"] = map[string]interface{}{
+			"podSecurityPolicyTemplate": actionInput["podSecurityPolicyTemplate"],
+		}
+	} else {
+		if statusAsMap, ok := status.(map[string]interface{}); ok {
+			statusAsMap["podSecurityPolicyTemplate"] = actionInput["podSecurityPolicyTemplate"]
+		} else {
+			return fmt.Errorf("error looking up podSecurityPolicyTemplate")
+		}
+	}
+
+	_, err = request.Schema.Store.Update(request, request.Schema, project, request.ID)
+	if err != nil {
+		return fmt.Errorf("error updating project: %v", err)
+	}
+
+	return nil
+}
+
+func (h *Handler) createNewBinding(request *types.APIContext, schema *types.Schema,
+	actionInput map[string]interface{}) (map[string]interface{}, error) {
+	binding := make(map[string]interface{})
+	binding["projectId"] = request.ID
+	binding["podSecurityPolicyTemplate"] = actionInput["podSecurityPolicyTemplate"]
+
+	name := fmt.Sprintf("%v-%v-pspt-binding", strings.Replace(request.ID, ":", "-",
```

Comment:
Dont set name or id. The api framework needs to take care of setting the ID and this resource does not need a name (it doesnt really have a name field because the Name that is store in k8s is our id field, it gets translated).